src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        if (Array.isArray(action.keysToRemove)) {
+          action.keysToRemove.forEach((key) => {
+            delete currentState[key];
+          });
+        }
+        break;
+      default:
```

Comment:
The default case in the switch statement is empty. While it's good that you have a default case for error handling, it would be beneficial to include some logic here, such as throwing an error or logging a message when an action with an unrecognized type is encountered.